File: frontend/src/shared/api/book/types/book.ts
export type BookType = 'Paper' | 'Electronic' | 'Audio' | string;

export type BookFormat = {
  id: number;
  formatType: BookType;
  isAvailable: boolean;
  pagesCount: number;
  durationMinutes: number;
};

export type BookFormatPayload = {
  formatType: BookType;
  pagesCount: number;
  durationMinutes: number;
};

// Оновлена модель книги, яку ми отримуємо з сервера (Response)
export type Book = {
  id: number;
  title: string;
  author: string;
  description: string;
  genreName: string;
  language: string;
  publicationYear: number;
  formats: BookFormat[]; // Тепер тут масив об'єктів
  coverUrl: string | null;
  downloadUrl: string | null;
  listenUrl: string | null;
  createdAt: string; // Сервер повертає ISO-рядок. Можна змінити на Date, якщо ти робиш new Date() при отриманні
};

// Оновлений Payload для створення/редагування книги (Request)
export type BookPayload = {
  title: string;
  author: string;
  description: string;
  genreName: string;
  language: string;
  publicationYear: number;
  formats: BookFormatPayload[]; // Відправляємо масив форматів без id та isAvailable
};

File: backend/DigitalLibrary.DAL/Entities/Book.cs
namespace DigitalLibrary.DAL.Entities;

public class Book
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Author { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public int GenreId { get; set; }

    public Genre Genre { get; set; } = null!;

    public string Language { get; set; } = string.Empty;

    public int PublicationYear { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public List<BookFormat> Formats { get; set; } = new();

    public List<BookFile> Files { get; set; } = new();
}